### PR TITLE
Reroll cost improvements

### PR DESCRIPTION
**Pull request by EllH**
Improved reroll costs to have stack and single values and added cost to pity row, changed reroll tracker to use user data store and fixed a few small issues with it 
image

### CODE REVIEW

File: src/stores/UserData.ts [hunk 1]
```diff
@@ -113,13 +115,36 @@ export const useUserDataStore = defineStore('userDataStore', () => {
...
         return JSON.parse(localStorage.getItem('rerollCost') ?? '{}')
     }
 
+    function getRerollTracker(): RerollTrackerInterface {
+        // Handle Old Local Data
+        if(localStorage.getItem('mod_reroll_tracker')) {
+            const oldRerollCount = localStorage.getItem('mod_reroll_tracker');
+            const newRerollCount = { currentCount: oldRerollCount ? parseInt(oldRerollCount) : 0 }
+            localStorage.removeItem('mod_reroll_tracker');
+            localStorage.setItem('rerollTracker', JSON.stringify(newRerollCount))
```
Comment (EllH) [inline src/stores/UserData.ts:133]: the new reroll format won't save if someone loads the page and doesn't change the reroll tracker without this

**Comment by kabamgamer** [comment]
@EllH , thanks for another PR.
Could you please resolve merge conflicts, I made a few changes this morning.

**Review by kabamgamer — CHANGES_REQUESTED** [review]
Please fix merge conflicts

File: src/components/sections/RerollCost.vue [hunk 2]
```diff
@@ -29,7 +34,14 @@
             
             <tr>
               <th scope="row">Cost per 286</th>
-              <td scope="row" v-for="calculateCosts in getCalculateCostForRerollCount(286)" :key="calculateCosts">{{ calculateCosts }}</td>
+              <td scope="row" v-for="calculateCosts in getCalculateCostForRerollCount(pitySystemLimit)" :key="calculateCosts">{{ calculateCosts }}</td>
```
Comment (EllH) [inline src/components/sections/RerollCost.vue:37]: @kabamgamer these keys aren't unique in a decent chunk of scenarios

**Comment by EllH** [comment]
> Please fix merge conflicts

fixed

File: src/components/sections/RerollCost.vue (hunk 2 above)
Comment (EllH) [reply src/components/sections/RerollCost.vue:37]: easiest one to repo would be 0 as values

Comment (EllH) [reply src/components/sections/RerollCost.vue]: I've fixed this in my latest commit, slightly changed how we build the table to having a list of "chaosTiers" with title, mote amount and gold amount, using row prefix + title

**Review by kabamgamer — APPROVED** [review]

**Pull request merged**